11-functions.Rmd

```diff
@@ -5,6 +5,123 @@
 ottrpal::set_knitr_image_path()
 ```
 
-```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
+```{r, fig.align='center', fig.alt = "Learning objectives are to be able to: Recognize when it is appropriate to make a custom function to aid in the reproducibility of an analysis. Create functions that are reproducible and useable by others. What aspects of a function make them useable by others. Well documented. Follow a style. Properly engineered", out.width="100%", echo = FALSE}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_53")
 ```
+
+### Recognize when to write a custom function
+
+There's a lot of reasons writing your own custom functions might be a thing you need to do.
+
+#### Cutting down on repetitive code
+
+We've discussed how DRY code is easier to maintain and understand! It is easier to fix code in one place as opposed to three or more places. A general guideline is that if you need two use a same chunk of code more than twice you probably should invest the time to make a custom function for it.
+
+```{r, fig.align='center', fig.alt = "The parrot is saying 'I’m using this code chunk three times. Updating this might get hairy… Good time to make a custom function!' The computer has three identical chunks of code on the screen.
+", out.width="100%", echo = FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g341450bc187_100_5")
+```
+
+#### More readable code
+
+Custom functions can allow you to organize code into manageable chunks.
+By creating custom functions it can be easier to organize our code in a way thats more readable.
+
+
+### Writing a function
+
+A common way that a custom function might come about is:
+
+1. You've written some initial code that takes some object as input.
+2. You now realize you will need to run this series of code more than twice.
+
+
+#### Starting a new function
+
+To get started with writing a new function, you can follow these steps to make a custom function:
+
+1. Open up a new R script file to copy and paste our basic template we have below.
+```
+#' Title here
+#'
+#' @description This is what the function does
+#'
+#' @param arg1 A first argument
+#' @param arg2 A second argument
+#'
+#' @return What is returned
+#' @export
+#'
+#' @examples
+#'
+#' function(arg1, arg2)
+#'
+#'
+function_name <- function(arg1, arg2) {
+
+    # Your main code here
+
+    return(output)
+}
+```
+Your basic custom function is going to follow this type of structure:
+
+We will walk through how to use this template here.
+
+This top part of the template is the documentation. It's formatted in a way that would
```

Comment:
```suggestion
This top part of the template is the documentation. It's formatted in a way that would allow you to incorporate it into a package later if desired.
```

Reply:
just adjusting for readability and adding to the same line